### PR TITLE
Add bunch of test cases

### PR DESCRIPTION
**Pull request by kant2002**
- Add both working, and not yet working snippets of C
- Add support for recursive dir traversal in integration testing. Will need this for more tests to C runtime

### CODE REVIEW

**Comment by kant2002** [comment]
At least 1 test should be removed after #185 lands, since I have basically duplicate test.

File: Cesium.Compiler/stdlib/stdio.h
Comment (ForNeVeR) [inline Cesium.Compiler/stdlib/stdio.h:8]: I'm not sure if it is valid for `stdio.h` to define the function body in a header file.

Also, this deserves a TODO comment (I'm creating an issue for each merged `TODO`).

File: Cesium.Runtime/StdLibFunctions.cs
Comment (ForNeVeR) [inline Cesium.Runtime/StdLibFunctions.cs:12]: Wow that's a gimmick. Let's write a unit test for the standard library, then?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
Also, I'm not merging this until everything's green. Have you investigated the reason for the test failures? Do you need help?

File: Cesium.Runtime/StdLibFunctions.cs
Comment (kant2002) [reply Cesium.Runtime/StdLibFunctions.cs:12]: Maybe we can, but I mostly think that should do that as part of "acceptance testing", but in retrospection after I think a bit on this I do agree with you. That's subsystem should be tested.

File: Cesium.Compiler/stdlib/stdio.h
Comment (kant2002) [reply Cesium.Compiler/stdlib/stdio.h:8]: Okay. that PR is kitchen sink of different failed attempts to find some place where I can have quick win. I will try to cleanup it which is quite possible, but still it reply on earlier RPs which I send, so I waiting for them to land. I will let you know when this would be ready for your attention.

**Comment by kant2002** [comment]
Abs is parsing INT_MIN. Fixed.
printf is having float point constants as part of tree, If's in progress now

**Comment by kant2002** [comment]
Ooh, abs is also requires nested `#include` because  I add `#include <math.h>`

File: Cesium.Compiler/stdlib/stdio.h
Comment (ForNeVeR) [reply Cesium.Compiler/stdlib/stdio.h:8]: For such cases, you may consider marking a PR as a "draft". This way, you formally let me know that it is not yet ready to merge.

An example of a draft PR is #139.

Comment (kant2002) [reply Cesium.Compiler/stdlib/stdio.h:8]: Yeah, I was lazy when I create it, but cannot convert PR to draft after creation. Maybe that's some permission issue, or require some setup?

Comment (ForNeVeR) [reply Cesium.Compiler/stdlib/stdio.h:8]: Sorry, I'm not sure what the rules about drafts are, but I've found that I can convert it myself. So I did.

Let me know if you want to un-draft this, and cannot do it yourself.

**Comment by kant2002** [comment]
Now it's waiting for #189 only

File: Cesium.Compiler/stdlib/stdio.h
Comment (ForNeVeR) [reply Cesium.Compiler/stdlib/stdio.h:8]: What's your opinion on this `printf` implementation? Should we still merge it being in a header file?

Comment (kant2002) [reply Cesium.Compiler/stdlib/stdio.h:8]: I would like to have `Cesium.Runtime` to be something minimal and provide as much implementation in C as possible. Example of functions which maybe written in C is `atoi`/`ioa`, we almost at the point where we can write them in C. Maybe that's require have something like `stdlib.c` which would be added automatically by compiler. That's just example, not something which I think though.

For marketing reasons, I think we should have `printf` even with limitation as serious as it is right now. Technically right now we don't have "Hello World".

Comment (ForNeVeR) [reply Cesium.Compiler/stdlib/stdio.h:8]: My whole point is that in C, you cannot write code in headers.

If we want to write C stdlib (how do you distinguish it from `Cesium.Runtime`, by the way?), then we'll have to do it in C files, and link them somehow with the user's program.

For now, this is too much. Let's migrate this to the `Cesium.Runtime`, and create a separate issue to discuss details of implementation parts of the standard library in C.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**
